x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/frozen_indices_handling/trial_license_complete_tier/frozen_indices_handling.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import expect from '@kbn/expect';
+import { CreateRuleProps } from '../../../../../../../api_integration/services/security_solution_api.gen';
+
+import { dataGeneratorFactory } from '../../../../utils';
+import {
+  deleteAllRules,
+  deleteAllAlerts,
+} from '../../../../../../../common/utils/security_solution';
+import { FtrProviderContext } from '../../../../../../ftr_provider_context';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const log = getService('log');
+  const securitySolutionApi = getService('securitySolutionApi');
+  const retry = getService('retry');
+
+  describe('@ess frozen indices queried', () => {
+    const index = 'test_idx';
+    const { indexListOfDocuments } = dataGeneratorFactory({ es, index, log });
+
+    const cleanOperationsLIFO: Array<() => Promise<void>> = [];
+    const addCleanOperation = (operation: () => Promise<void>) =>
+      cleanOperationsLIFO.push(operation);
+    const runCleanOperations = async () => {
+      for (let idx = cleanOperationsLIFO.length - 1; idx >= 0; idx--) {
+        await cleanOperationsLIFO[idx]();
+      }
+    };
+
+    after(async () => {
+      await deleteAllAlerts(supertest, log, es);
+      await deleteAllRules(supertest, log);
+      await runCleanOperations();
+    });
+
+    it('should report the querying of frozen indices during rule execution', async () => {
+      const id = uuidv4();
+      const timestamp = new Date().toISOString();
+      const firstDoc = {
+        id,
+        '@timestamp': timestamp,
+        foo: 'bar',
+      };
+
+      await indexListOfDocuments([firstDoc]);
+
+      const createRuleProps: CreateRuleProps = {
+        body: {
+          name: 'spammy_1s',
+          description: 'Spammy query rule',
+          enabled: true,
+          risk_score: 1,
+          rule_id: 'rule-1',
+          severity: 'low',
+          type: 'query',
+          query: 'foo: *',
+          index: ['test_idx*'],
+          interval: '1s',
```

Comment:
Instead of a `1s` interval, there's an alerting framework [run soon](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/alerting/server/routes/run_soon.ts) API that we can use. Disabling/re-enabling the rule also works. Or, if we split the test into multiple tests and ensure that the data is set up prior to creating and enabling the rule, then the rule will only need to run once.

`1s` interval is technically valid, but never recommended to customers in production and slightly raises the likelihood of test flakiness from random problems with the low interval. In production, `1m` would be considered a low interval and `10s` would be extremely low. I thought the framework had a minimum interval setting, but maybe not if `1s` was allowed?